packages/react-make-styles/src/makeStyles.test.tsx

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+import { makeStyles } from './makeStyles';
+
+describe('makeStyles', () => {
+  it('works outside of React components', () => {
+    expect(() => makeStyles({ root: { color: 'red' } })).not.toThrow();
+  });
+
+  it('throws inside React components', () => {
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    jest.spyOn(console, 'error').mockImplementation(() => {});
```

Comment:
This is a leftover and thus should be removed right ?

Reply:
Nope, even if `expect().toThrow()` catches errors, React still logs (even with error boundaries) an error to console, https://github.com/facebook/react/issues/15069.

```
at renderRoot (office-ui-fabric-react\node_modules\react-dom\cjs\react-dom.development.js:19468:13) Error: make
Styles(): this function cannot be called in component's scope. All makeStyles() calls should be top level i.e. in a root scope of a file.
at Object.makeStyles (office-ui-fabric-react\packages\react-make-styles\src\makeStyles.ts:23:13)
at Example (C:\Users\olfedias\WebstormProjects\office-ui-fabric-react\packages\react-make-styles\src\makeStyles.test.tsx:22:7)
```

As we have spies for `console.*` calls in Jest setup: https://github.com/microsoft/fluentui/blob/8b08552fe428f48dc17b7444fa186de750d380e7/scripts/jest/setupTests.js#L22-L27

This will cause a failure of this test 🙄

Reply:
oh! thanks for explanation 🙌 I'd recommend to get rid of that magic for convergence ?

- as a followup it would be great to add this explanation to this line, otherwise one can spent ages on troubleshooting why something doesnt work as expected 😅

Reply:
> I'd recommend to get rid of that magic for convergence ?

I _like_ & I _don't like_ this logic:
- previously it saved us multiple times when tests were passing, but React was logging errors
- it helped to detect all places with React deprecations before

Northstar has a nicer version of this and allows to use `consoleUtil.disableOnce()` for such cases:
https://github.com/microsoft/fluentui/blob/master/packages/fluentui/react-northstar/test/utils/consoleUtil.ts

Reply:
speaking from experience, what we need in general is a small logging abstraction that is configurable, thus  user has much better ergonomics.

also if one want's to suppress console logs from underlying abstraction (react) it should be done per test, to mitigate magic factor